core/src/main/kotlin/org/jetbrains/kotlinx/dataframe/api/KeyValueProperty.kt

```diff
@@ -0,0 +1,14 @@
+package org.jetbrains.kotlinx.dataframe.api
+
+import org.jetbrains.kotlinx.dataframe.annotations.ColumnName
+import org.jetbrains.kotlinx.dataframe.annotations.DataSchema
+
+/** A [DataSchema] interface / class can implement this if it represents a map-like data schema (so key: value). */
+public interface KeyValueProperty<T> {
+    // needs to be explicitly overridden in @DataSchema interface, otherwise extension functions won't generate (TODO)
```

Comment:
this interface can be annotated with @DataSchema itself. then extensions going to be a part of the library API and could be called on every marker that implements this interface

Reply:
sure, why not :)